website/docs/r/private_link_service.html.markdown

```diff
@@ -0,0 +1,136 @@
+  location            = azurerm_resource_group.example.location
+
+  auto_approval_subscription_ids              = ["00000000-0000-0000-0000-000000000000"]
+  visibility_subscription_ids                 = ["00000000-0000-0000-0000-000000000000"]
+  load_balancer_frontend_ip_configuration_ids = [azurerm_lb.example.frontend_ip_configuration.0.id]
+
+  nat_ip_configuration {
+    name                       = "primaryIpConfiguration"
+    private_ip_address         = "10.5.1.17"
+    private_ip_address_version = "IPv4"
+    subnet_id                  = azurerm_subnet.example.id
+    primary                    = true
+  }
+
+  nat_ip_configuration {
+    name                       = "secondaryIpConfiguration"
+    private_ip_address         = "10.5.1.18"
+    private_ip_address_version = "IPv4"
+    subnet_id                  = azurerm_subnet.example.id
+    primary                    = false
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the private link service. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the resource group in which the private link service resides. Changing this forces a new resource to be created.
+
+* `location` - (Required) Specifies the supported Azure location where the resource exists. Changing this forces a new resource to be created.
+
+* `auto_approval_subscription_ids` - (Optional) A list of subscription globally unique identifiers(GUID) that will be automatically be able to use this service.
+
+* `visibility_subscription_ids` - (Optional) A list of subscription globally unique identifiers(GUID) that will be able to see this service. If left undefined all Azure subscriptions will be able to see this service.
+
+* `nat_ip_configuration` - (Required) A `nat_ip_configuration` block as defined below.
+
+* `load_balancer_frontend_ip_configuration_ids` - (Required) A list of Standard Load Balancer(SLB) resource IDs. The Private Link service is tied to the frontend IP address of a SLB. All traffic destined for the private link service will reach the frontend of the SLB. You can configure SLB rules to direct this traffic to appropriate backend pools where your applications are running.
+
+* `tags` - (Optional) A mapping of tags to assign to the resource. Changing this forces a new resource to be created.
+
+---
+
+The `nat_ip_configuration` block supports the following:
+
+* `name` - (Required) The name of primary private link service NAT IP configuration. Changing this forces a new resource to be created.
+
+* `private_ip_address` - (Optional) The private IP address of the NAT IP configuration.
+
+* `private_ip_address_version` - (Optional) The ip address version of the `ip_configuration`, the supported value is `IPv4`. Defaults to `IPv4`.
+
+-> **NOTE:** Private Link Service Supports `IPv4` traffic only.
+
+* `subnet_id` - (Required) The resource ID of the subnet to be used by the service.
```

Comment:
COuld we add a note about how `enforce_private_link_service_network_policies` needs to be set to true on the subneet?

Reply:
Fixed.